src/main/kotlin/io/csbroker/apiserver/service/post/PostServiceImpl.kt

```diff
@@ -0,0 +1,86 @@
+package io.csbroker.apiserver.service.post
+
+import io.csbroker.apiserver.common.enums.ErrorCode
+import io.csbroker.apiserver.common.exception.EntityNotFoundException
+import io.csbroker.apiserver.common.exception.UnAuthorizedException
+import io.csbroker.apiserver.controller.v1.post.response.CommentResponseDto
+import io.csbroker.apiserver.controller.v1.post.response.PostResponseDto
+import io.csbroker.apiserver.model.Post
+import io.csbroker.apiserver.model.PostLike
+import io.csbroker.apiserver.repository.post.CommentRepository
+import io.csbroker.apiserver.repository.post.PostLikeRepository
+import io.csbroker.apiserver.repository.post.PostRepository
+import io.csbroker.apiserver.repository.problem.ProblemRepository
+import io.csbroker.apiserver.repository.user.UserRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional(readOnly = true)
+class PostServiceImpl(
+    private val problemRepository: ProblemRepository,
+    private val postRepository: PostRepository,
+    private val postLikeRepository: PostLikeRepository,
+    private val userRepository: UserRepository,
+    private val commentRepository: CommentRepository,
+) : PostService {
+    override fun findByProblemId(problemId: Long, email: String?): List<PostResponseDto> {
+        val problem = problemRepository.findByIdOrNull(problemId) ?: throw EntityNotFoundException(
+            "${problemId}번 문제는 존재하지 않습니다",
+        )
+        val posts = postRepository.findAllByProblem(problem)
+        val comments = commentRepository.findAllByPostIn(posts)
+        val postLikes = postLikeRepository.findAllByPostIn(posts)
+        return posts.map {
+            PostResponseDto(
+                id = it.id!!,
+                content = it.content,
+                username = it.user.username,
+                likeCount = postLikes.count { postLike -> postLike.post == it }.toLong(),
+                isLiked = postLikes.any { postLike -> postLike.post == it && postLike.user.email == email },
+                comments = comments.filter { comment -> comment.post == it }.map { comment ->
+                    CommentResponseDto(
+                        id = comment.id!!,
+                        content = comment.content,
+                        username = comment.user.username,
+                        createdAt = comment.createdAt!!,
+                    )
```

Comment:
```suggestion
                    CommentResponseDto(comment)
```
중복코드를 피하도록 파싱하는 부분은 DTO한테 위임하는게 어떄요?

Reply:
넵 반영했습니당.